### PR TITLE
Mitm bug, install deps message issue

### PR DESCRIPTION
**Pull request by blakebyrnes**
2 bug fixes
1. The message to run the 'install dependencies' script broke from showing the original message in core. Writing a unit test for this led to me to an unhandled rejection issue.
2. When the mitm re-uses http1 connections, they can disconnect before they get reused, but the socket doesn't know. This fix tests out the socket before reusing it (by attempting to send the http request and seeing if an econnreset happens)

### CODE REVIEW

File: client/connections/ConnectionToCore.ts [hunk 1]
```diff
@@ -181,15 +181,15 @@ export default abstract class ConnectionToCore extends TypedEventEmitter<{
...
+  public async createSession(options: ICreateSessionOptions): Promise<CoreSession | Error> {
     try {
       const sessionMeta = await this.commandQueue.run<ISessionMeta>('Session.create', options);
       const session = new CoreSession({ ...sessionMeta, sessionName: options.sessionName }, this);
       this.coreSessions.track(session);
       return session;
     } catch (error) {
       if (error instanceof DisconnectedFromCoreError && this.isDisconnecting) return null;
-      throw error;
+      return error;
```
Comment (calebjclark) [inline client/connections/ConnectionToCore.ts:192]: Is this a new pattern? Returning errors instead of throwing them?

Comment (blakebyrnes) [reply client/connections/ConnectionToCore.ts:192]: It's not new... this is part of the unfortunate reality that if you throw this error and no one is looking, node shits itself.

Comment (calebjclark) [reply client/connections/ConnectionToCore.ts:192]: But you're catching the errors down in the Agent class.

Comment (blakebyrnes) [reply client/connections/ConnectionToCore.ts:192]: I was trying to track down an unhandled exception. I can switch this back and re-catch in Agent if you want to minimize locations we do this weird thing.

File: commons/test/TypeSerializer.test.ts [hunk 2]
```diff
@@ -2,9 +2,9 @@ import Puppet from '@secret-agent/puppet';
 import Chrome83 from '@secret-agent/emulate-chrome-83';
 import TypeSerializer, { stringifiedTypeSerializerClass } from '../TypeSerializer';
 import { CanceledPromiseError } from '../interfaces/IPendingWaitEvent';
-import Log from '../Logger';
+import logger from '../Logger';
```
Comment (calebjclark) [inline commons/test/TypeSerializer.test.ts:5]: Doesn't this go against our convention of naming the import variable the same name as the file?

Comment (blakebyrnes) [reply commons/test/TypeSerializer.test.ts:5]: Yeah.. I exported the class "Log" from this class, and suddenly eslint started complaining about 3 files where we did things like this. All our logger imports go against several conventions. We use this code everywhere, but the default export is actually a logger function, which should be named createLogger. I think at some point we just need a quick cleanup of the logger exports and imports.

`import Log from '@secret-agent/commons/Logger';`

**Pull request merged**